git_machete/git_operations.py

```diff
@@ -645,7 +646,7 @@ def __get_merge_base(self, sha1: FullCommitHash, sha2: FullCommitHash) -> FullCo
             #   then there is exactly one merge-base - the ancestor,
             # * if neither of sha1, sha2 is an ancestor of another,
             #   then none of the (possibly more than one) merge-bases is equal to either of sha1/sha2 anyway.
-            self.__merge_base_cached[sha1, sha2] = FullCommitHash.of(self._popen_git("merge-base", sha1, sha2).rstrip())
+            self.__merge_base_cached[sha1, sha2] = FullCommitHash.of(self._popen_git("merge-base", sha1, sha2, allow_non_zero=True).rstrip())
```

Comment:
Actually... in the zero merge-base case, it should probably be `None` saved into `self.__merge_base_cached[sha1, sha2]`... currently it'd be `FullCommitHash.of("")`

Reply:
> Also, I'd extract the first line from the output of git merge-base... in case of criss-cross histories, there will be >1 line in the output, and we shouldn't wrap this multiline string into FullCommitHash

There is no need for that because `merge-base`, by default returns only one merge base, you would have to provide `--all` option in order to get all merge bases.

Reply:
> Actually... in the zero merge-base case, it should probably be None saved into self.__merge_base_cached[sha1, sha2]... currently it'd be FullCommitHash.of("")

Yes, None is already saved into `self.__merge_base_cached[sha1, sha2]`, because `FullCommitHash.of("")` returns None.

Reply:
Oh! okay, sorry I didn't remember the former and wasn't aware of the latter 😅 let's leave as is then